OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/menu/OpenTemplate.java

```diff
@@ -144,6 +144,14 @@ private Template getTemplateFromElement(NodeList templateNodes) {
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        this.setVisible(true);
+        Enum canOpen = etc.canOpen();
+        
+        if (canOpen == EditorTopComponent.CanOpenState.SAVE_AND_OPEN) {
+            etc.fileAction().saveFile();
+            this.setVisible(true);
+            
+        } else if (canOpen == EditorTopComponent.CanOpenState.OPEN) {
+            this.setVisible(true);
```

Comment:
Take a look at this:
![switch_case](https://cloud.githubusercontent.com/assets/17130832/15371089/2d064b88-1d3a-11e6-8840-558de5c98aed.png)
Is it really necessary to throw an exception? Or even implement default case? canOpen() can't return anything different than SAVE_AND_OPEN, OPEN, CANCEL.